identities/src/main/java/org/commonjava/maven/atlas/ident/util/VersionUtils.java

```diff
@@ -108,4 +108,26 @@ public static SingleVersion createSingleVersion( final String version )
         }
     }
 
+    public static boolean isValidSingleVersion( final String version )
+    {
+        boolean result = true;
+        if ( version == null || version.trim().length() < 1 )
+        {
+            result = false;
```

Comment:
I'd just go ahead and return false here. The logic you have here doesn't prevent trying to parse when the input value is null or empty.

And when you have that fixed, go ahead and merge at will. :smiley: